### PR TITLE
feat: use an editor to write the change summary

### PR DESCRIPTION
**Pull request by zkochan**
ref #346

The input prompt of enquirer is terrible for writing long texts.
I think changeset should use the favorite editor of the user for describing the changes.

### CODE REVIEW

**Comment by mitchellhamilton** [comment]
Thanks for this @zkochan! Could you change this so that it only uses an editor when an empty string is returned from enquirer so for users who want to use an editor, they can press enter once more but people can also write the message without using an editor by default. The reason for this is that vim is the default value of `EDITOR` on macOS IIRC which could really confuse people who are unfamiliar with it. Also, can you handle the errors that `external-editor` can throw and fallback to enquirer?

**Comment by zkochan** [comment]
ok

**Comment by Andarist** [comment]
@MichaelKapustey could u sign the CLA?

**Comment by MichaelKapustey** [comment]
Did that.

**Comment by MichaelKapustey** [comment]
I've signed CLA with my github related email. But push is somehow using my work email. I'm not able to update it through the CLA form.

**Comment by MichaelKapustey** [comment]
Hey guys. CLA check is acting weird. I've already subscribed it using my github email. Figured out that my work email was in the commit. Tried to subscribe CLA once again with work email and it didn't allow me to do that. But CLA check started to pass. Now Zoltan merged master to make branch up to date and CLA check fails on my commit again.

**Comment by Andarist** [comment]
:sob: any way to debug this @mitchellhamilton ? or should we maybe reset history on this branch to avoid this altogether?

File: .changeset/modern-ligers-fetch.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@changesets/cli": minor
+---
+
+The change summary is typed through an editor, not through an input prompt in the terminal.
```
Comment (mitchellhamilton) [inline .changeset/modern-ligers-fetch.md:5]:
```suggestion
Open an editor when no message is given when creating a changeset
```

**Pull request merged**